src/shmem_team.c

```diff
@@ -177,6 +178,7 @@ int shmem_internal_team_init(void)
 
     /* Initialize an integer used to agree on an equal return value across PEs in team creation: */
     team_ret_val = shmem_internal_shmalloc(sizeof(int) * 2);
+    if (NULL == team_ret_val) return -1;
```

Comment:
If `team_init` fails, the calling context does not call `team_fini`, so we should really be trying to do the cleanup here before returning from init. I will push a patch to show you what I mean.

Reply:
Oh yeah - that makes sense.  Thanks!